evm/src/cpu/kernel/asm/curve/bn254/curve_arithmetic/pairing.asm

```diff
@@ -0,0 +1,154 @@
+/// def bn254_pairing(pairs: List((Curve, TwistedCurve))) -> Bool:
+///     
+///     for P, Q in pairs:
+///         if not (P.is_valid and Q.is_valid):
+///             return @U256_MAX
+///     
+///     out = 1
+///     for P, Q in pairs:
+///         out *= miller_loop(P, Q)
+///
+///     result = bn254_final_exponent(out)
+///     return result == @GENERATOR_PAIRING
+
+global bn254_pairing:
+    // stack: 
```

Comment:
So is the expected stack empty when calling this?

Reply:
this happened by accident after cutting and pasting events 😂